python/tvm/relay/op/contrib/dnnl.py

```diff
@@ -207,7 +211,205 @@ def pattern_table():
     return dnnl_patterns
 
 
-def partition_for_dnnl(mod, params=None):
+def get_optimal_layout_for_conv(input_size, weight_shape, out_shape, paddings, strides, dilates, G):
+    """Get the optimal layout of dnnl, given shape of conv2d.
+
+    Parameters
+    ----------
+    input_size, weight_shape, out_shape, paddings, strides, dilates, G : Int, String
+                                                                         Input argument.
+
+    Returns
+    -------
+    layouts : string
+              The result.
+    """
+    return _ffi_api.get_optimal_layout_for_conv(
+        input_size,
+        weight_shape,
+        out_shape,
+        paddings,
+        strides,
+        dilates,
+        G,
+    )
+
+
+def get_optimal_layout_for_conv_transpose(
+    input_size, weight_shape, out_shape, paddings, output_paddings, strides, dilates, G
+):
+    """Get the optimal layout of dnnl, given shape of tranposed conv2d.
+
+    Parameters
+    ----------
+    input_size, weight_shape, out_shape, paddings, output_paddings, strides, dilates, G
+        : Int, String
+          Input argument.
+
+    Returns
+    -------
+    layouts : string
+              The result.
+    """
+    return _ffi_api.get_optimal_layout_for_conv_transpose(
+        input_size,
+        weight_shape,
+        out_shape,
+        paddings,
+        output_paddings,
+        strides,
+        dilates,
+        G,
+    )
+
+
+def get_shape(tensor):
+    """Get tensor's shape."""
+    if isinstance(tensor, relay.expr.Var):
+        return tensor.type_annotation.concrete_shape
+    elif isinstance(tensor, relay.expr.Constant):
+        return tensor.data.shape
+    elif isinstance(tensor, tvm.ir.tensor_type.TensorType):
+        return tensor.concrete_shape
+    elif isinstance(tensor, tvm.ir.container.Array):
+        return tensor[-1].shape
+    elif isinstance(tensor, relay.expr.Call):
+        return tensor.checked_type.shape
+    else:
+        raise TypeError("Unsupport data type: %s" % type(tensor))
+
+
+def validate_layout_for_tvm(input_data, is_weight=False, conv_type=1):
```

Comment:
suggest to remove "_for_tvm" from the func name, and `conv_type` to a list of something like `"conv1d", "conv2d", `conv3d"` for better readibility?

Reply:
This function are translating dnnl tag to tvm layout, need a better name to reflect this.

Reply:
change this name from `validate_layout_for_tvm` into `tag2layout`.